### ISSUE TITLE
iconv on Alpine 3.13

### ISSUE THREAD

**Issue by enumag** [docker-library/php#1121]
Using the PHP alpine images, this code...:

```php
<?php
ini_set('display_errors', '1');
error_reporting(-1);
iconv("UTF-8", "ASCII//TRANSLIT", "foobar");
```

...causes this error:

```
iconv(): Wrong charset, conversion from `UTF-8' to `ASCII//TRANSLIT' is not allowed in [...]
```

This is a long-known [issue](https://github.com/docker-library/php/issues/240), however the situation has changed significantly so I think a new issue is appropriate.

Until now the problem was commonly solved by this workaround:

```
RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ gnu-libiconv
ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so
```

This however no longer works with gnu-libiconv 1.16-r0 because the `preloadable_libiconv.so` has been removed (see [alpine aports issue](https://gitlab.alpinelinux.org/alpine/aports/-/issues/12328) and [related libiconv commit](https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=commit;h=f38356d4414c023fcfbe34934d60c08ee94d5b8c)).

On Alpine 3.12 we can use `gnu-libiconv=1.15-r2`.

~~On Alpine 3.13 however the version no longer exists and consequently the workaround is not available.~~

EDIT: My bad. The version is available in 3.13, just missing in edge.

Is there any other way to fix this? Perhaps somehow using `--with-iconv` when building PHP? (Not sure, just an idea.)

**Comment by khs1994** [comment]
```docker
RUN apk add --no-cache  gnu-libiconv
ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so
```

**Comment by khs1994** [comment]
* https://pkgs.alpinelinux.org/contents?branch=v3.13&name=gnu-libiconv&arch=x86_64&repo=community
* https://pkgs.alpinelinux.org/contents?branch=edge&name=gnu-libiconv&arch=x86_64&repo=community

**Comment by enumag** [comment]
Ah okay so it only doesn't work with edge for now but v3.13 will be fine. Good to know but still it means that the current workaround might not work later.

**Comment by greg0ire** [comment]
To anyone visiting this issue, the current workaround is

```diff
-apk add gnu-libiconv
+apk add gnu-libiconv=1.15-r2
```

**Comment by gouki777** [comment]
I looked at the documentation , used 3.12 is good.  thanks~
```
FROM alpine:3.12
RUN     apk update && \
        apk add php7=7.3.27-r0 php7-iconv 
		.............................		
#fix apline-php7-iconv
RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ gnu-libiconv=1.15-r2
ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
cmd["php"]
```

**Comment by litan1106** [comment]
my current workaround is the stick with alpine3.13 and gnu-libiconv=1.15-r3

**Comment by thekid** [comment]
> my current workaround is the stick with alpine3.13 and gnu-libiconv=1.15-r3

For easy copy&pasting:

```Dockerfile
RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/v3.13/community/ gnu-libiconv=1.15-r3

ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
```

**Comment by IStranger** [comment]
I got the same problem on Alpine **3.14** with php7.4. However, `preloadable_libiconv.so` was removed from 3.14.
You can just install the same extensions from **3.13** repo:
```
RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/v3.13/community/ gnu-libiconv=1.15-r3
ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
```

**Comment by litan1106** [comment]
> I got the same problem on Alpine **3.14** with php7.4. However, `preloadable_libiconv.so` was removed from 3.14:
> https://pkgs.alpinelinux.org/contents?file=&path=&name=gnu-libiconv&branch=v3.14&repo=community&arch=x86_64
> 
> You can just install the same extensions from **3.13** repo:
> 
> ```
> RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/v3.13/community/ gnu-libiconv=1.15-r3
> ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
> ```
> 
> It works great!

are you using php-7.4 with alpine 3.14? then install the gnu-libiconv from 3.13 repo?

**Comment by IStranger** [comment]
> are you using php-7.4 with alpine 3.14? then install the gnu-libiconv from 3.13 repo?

Yes. I'm using `php:7.4.21-fpm-alpine3.14` image from hub.docker.com.

**Comment by josefsabl** [comment]
> I have the same problem on Alpine **3.14** with php7.4. However, `preloadable_libiconv.so` was removed from 3.14: https://pkgs.alpinelinux.org/contents?file=&path=&name=gnu-libiconv&branch=v3.14&repo=community&arch=x86_64
> 
> You can just install the same extensions from **3.13** repo:
> 
> ```
> RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/v3.13/community/ gnu-libiconv=1.15-r3
> ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
> ```
> 
> It works great!

I confirm it also works with 3.15 at this time.

**Comment by duiniwukenaihe** [comment]
> 



> > 我在使用 php7.4 的 Alpine **3.14 上**遇到了同样的问题。但是，`preloadable_libiconv.so`已经从 3.14 中删除：[https://pkgs.alpinelinux.org/contents?file=&path=&name=gnu-libiconv&branch=v3.14&repo=社区&arch=x86_64](https://pkgs.alpinelinux.org/contents?file=&path=&name=gnu-libiconv&branch=v3.14&repo=community&arch=x86_64)
> > 您可以从**3.13** repo 安装相同的扩展：
> > ```
> > RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/v3.13/community/ gnu-libiconv=1.15-r3
> > ENV LD_PRELOAD /usr/lib/preloadable_libiconv.so php
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 效果很好！
> 
> 我确认此时也适用于3.15。


Are you sure this will work on  Alpine 3.15?I tried it and it failed：
```
https://github.com/richarvey/nginx-php-fpm/issues/262
```

**Comment by jdreesen** [comment]
This should not be necessary anymore, as it has been fixed here: #1264

**Comment by duiniwukenaihe** [comment]
thanks  i know
